project/SparkBuild.scala

```diff
@@ -792,7 +792,6 @@ object TestSettings {
       "JAVA_HOME" -> sys.env.get("JAVA_HOME").getOrElse(sys.props("java.home"))),
     javaOptions in Test += s"-Djava.io.tmpdir=$testTempDir",
     javaOptions in Test += "-Dspark.test.home=" + sparkHome,
-    javaOptions in Test += "-Dspark.testing=1",
```

Comment:
This is still set in the Maven build though. While you could also remove it there, maybe simplest to leave this as-is.